### PR TITLE
Option to bypass GuzzleHttp\ClientInterface object for Authentication class

### PR DESCRIPTION
**Pull request by valigara**
According to our conversation in this [PR](https://github.com/jlevers/selling-partner-api/pull/184) -> [comment](https://github.com/jlevers/selling-partner-api/pull/184#issuecomment-1026602628)

### CODE REVIEW

File: lib/Authentication.php [hunk 1]
```diff
@@ -102,10 +106,13 @@ public function requestLWAToken(): array
...
-        $res = $this->client->post($this->lwaAuthUrl, [
-            \GuzzleHttp\RequestOptions::JSON => $jsonData,
-        ]);
+        
+        $lwaTokenRequestHeaders = [
+            'Content-Type' => 'application/json',
+        ];
+        $lwaTokenRequestBody = \GuzzleHttp\json_encode($jsonData);
+        $lwaTokenRequest = new Psr7\Request('POST', $this->lwaAuthUrl, $lwaTokenRequestHeaders, $lwaTokenRequestBody);
+        $res = $this->client->send($lwaTokenRequest);
```
Comment (jlevers) [inline lib/Authentication.php:115]: What's the purpose of these changes?

File: lib/Authentication.php [hunk 2]
```diff
@@ -27,6 +27,10 @@ class Authentication
     private $requestTime;
     private $signingScope = null;
 
+    /**
+     * 
+     * @var \GuzzleHttp\ClientInterface
+     */
```
Comment (jlevers) [inline lib/Authentication.php:33]: Could this become one line please?

```php
/** @var \GuzzleHttp\ClientInterface */
```

**Review by jlevers — CHANGES_REQUESTED** [review]

File: lib/Authentication.php (hunk 2 above)
Comment (valigara) [reply lib/Authentication.php:33]: sure -> updated

File: lib/Authentication.php (hunk 1 above)
Comment (valigara) [reply lib/Authentication.php:109]: to fit `\GuzzleHttp\ClientInterface` it has no `post` method that was used before,
also it gives more flexibility as benefit -> not only `GuzzleHttp\Client` object can be used now it can be anything that implements `GuzzleHttp\ClientInterface` interface

**Review by jlevers — CHANGES_REQUESTED** [review]
One tiny README typo, other than that looks good!

File: README.md [hunk 3]
```diff
@@ -86,6 +86,7 @@ The array passed to the `Configuration` constructor accepts the following keys:
 * `accessTokenExpiration (int)`: A Unix timestamp corresponding to the time when the `accessToken` expires. If `accessToken` is given, `accessTokenExpiration` is required (and vice versa).
 * `onUpdateCredentials (callable|Closure)`: A callback function to call when a new access token is generated. The function should accept a single argument of type [`SellingPartnerApi\Credentials`](https://github.com/jlevers/selling-partner-api/blob/main/lib/Credentials.php).
 * `roleArn (string)`: If you set up your SP API application with an AWS IAM role ARN instead of a user ARN, pass that ARN here.
+* `authenticationClient` (GuzzleHttp\ClientInterface)`: Optional `GuzzleHttp\ClientInterface` object that will be used to generate access token from refresh token
```
[the remark this answers is not in the document]
Comment (valigara) [reply README.md]: tnx! sorry for typo & delay -> non working days :)

File: lib/Authentication.php (hunk 1 above)
Comment (valigara) [reply lib/Authentication.php:109]: You are welcome!
